include/Extrema.h

```diff
@@ -0,0 +1,132 @@
+#ifndef __EXTREMA_H__
+#define __EXTREMA_H__
+
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Structure   :  Extrema_t
+// Description :  Data structure for finding the extrema
+//
+// Data Member :  Field  : Target field (e.g., _DENS or BIDX(DENS))
+//                Radius : Target radius
+//                Center : Target center
+//
+//                Value : Output extreme value
+//                Coord : Coordinates of the extrema
+//                Rank  : MPI rank of the extrema
+//                Level : AMR level of the extrema
+//                PID   : Local Patch index of the extrema
+//                Cell  : Cell indices within a patch of the extrema
+//
+// Method      :  Extrema_t     : Constructor
+//               ~Extrema_t     : Destructor
+//                CreateMPIType : Create a MPI derived datatype for struct Extrema_t
+//-------------------------------------------------------------------------------------------------------
+struct Extrema_t
+{
+
+// data members
+// --> must be consistent with CreateMPIType()
+// ===================================================================================
+// input parameters
+   int    Field;
+   double Radius;
+   double Center[3];
+
+// output parameters
+   real   Value;
+   double Coord[3];
+   int    Rank;
+   int    Level;
+   int    PID;
+   int    Cell[3];
+
+
+   //===================================================================================
+   // Constructor :  Extrema_t
+   // Description :  Constructor of the structure "Extrema_t"
+   //
+   // Note        :  Initialize the data members
+   //
+   // Parameter   :  None
+   //===================================================================================
+   Extrema_t()
+   {
+
+      Field     = -1;
```

Comment:
The initial value of `Field` should be 0 (`= _NONE`), otherwise the check [here](https://github.com/hyschive/gamer-fork/blob/pr-find-extrema/src/Auxiliary/Aux_FindExtrema.cpp#L82) has no effect.